sops/__init__.py

```diff
@@ -670,17 +683,22 @@ def walk_list_and_decrypt(branch, key, aad=b'', stash=None, digest=None):
             nstash = stash[i]
         if isinstance(v, dict):
             kl.append(walk_and_decrypt(v, key, aad=aad, stash=nstash,
-                                       digest=digest, isRoot=False))
+                                       digest=digest, isRoot=False, unencrypted=unencrypted))
         elif isinstance(v, list):
             kl.append(walk_list_and_decrypt(v, key, aad=aad, stash=nstash,
-                                            digest=digest))
+                                            digest=digest, unencrypted=unencrypted))
         else:
-            kl.append(decrypt(v, key, aad=aad, stash=nstash, digest=digest))
+            kl.append(decrypt(v, key, aad=aad, stash=nstash, digest=digest, unencrypted=unencrypted))
     return kl
 
 
-def decrypt(value, key, aad=b'', stash=None, digest=None):
+def decrypt(value, key, aad=b'', stash=None, digest=None, unencrypted=False):
     """Return a decrypted value."""
+    if unencrypted:
+        if digest:
+            digest.update(value)
+        return value
+
```

Comment:
humf.... looks like tangling the digest calculation with `decrypt()` wasn't a great idea after all...